include/accelerator.h

```diff
@@ -0,0 +1,80 @@
+#pragma once
+
+#include <invert_quda.h>
+
+namespace quda
+{
+
+  /**
+    @brief Generic accelerated solver with an accelerator. Let me explain what accelerate means here:
+    - Solvers are used in iterations based on an operator and a vector, e.g. to solver A * x = b, we have
+        b = Solver[A, x].
+    - Now assume we have another set of operator B and vector c and y. The acceleration works as
+        y = forward(x) -> c = Solver[B, y] -> b = backward(c),
```

Comment:
Perhaps we should make the analogy with multigrid language here explicit.  E.g., `forward` is the restrictor, `backward` is the prolongator and `Solver[B, ...]` is the coarse grid solve.

Reply:
https://github.com/lattice/quda/pull/1061/commits/377701f828ca406411764326773a4dcfdefd8b74